instance/tests/models/test_openedx_storage_mixins.py

```diff
@@ -275,6 +276,16 @@ def test_get_s3_connection(self):
         instance = OpenEdXInstanceFactory()
         s3_connection = instance.get_s3_connection()
         self.assertIsInstance(s3_connection, boto.s3.connection.S3Connection)
+        self.assertEqual(s3_connection.host, settings.AWS_S3_DEFAULT_HOSTNAME)
+
+        instance2 = OpenEdXInstanceFactory()
+        instance2.s3_region = 'eu-central-1'
+        s3_connection = instance2.get_s3_connection()
+        self.assertIsInstance(s3_connection, boto.s3.connection.S3Connection)
+        self.assertEqual(
+            s3_connection.host,
+            settings.AWS_S3_CUSTOM_REGION_HOSTNAME.format(region=instance2.s3_region)
+        )
```

Comment:
@lgp171188 If you override `settings.AWS_S3_DEFAULT_HOSTNAME` and `settings.AWS_S3_CUSTOM_REGION_HOSTNAME` via the decorator, it should be possible to reduce duplication here by using `ddt` to test different regions:

```python
    @override_settings(
        INSTANCE_STORAGE_TYPE=StorageContainer.S3_STORAGE,
        AWS_ACCESS_KEY='test',
        AWS_SECRET_ACCESS_KEY_ID='test',
        AWS_S3_DEFAULT_HOSTNAME='s3.test.host',
        AWS_S3_CUSTOM_REGION_HOSTNAME='s3.{region}.test.host',
    )
    @ddt.data(
        ('', 's3.test.host'),  # Default region (us-east-1)
        ('eu-central-1', 's3.eu-central-1.test.host'),
        # More regions ...
    )
    @ddt.unpack
    def test_get_s3_connection(self, s3_region, expected_hostname):
        """
        Test get_s3 connection returns right instance
        """
        instance = OpenEdXInstanceFactory()
        instance.s3_region = s3_region
        s3_connection = instance.get_s3_connection()
        self.assertIsInstance(s3_connection, boto.s3.connection.S3Connection)
        self.assertEqual(s3_connection.host, expected_hostname)
```

Reply:
Ah, this makes it cleaner. Will implement the changes for this :+1: